src/Symfony/Component/HttpKernel/EventListener/DumpListener.php

```diff
@@ -26,20 +27,28 @@ class DumpListener implements EventSubscriberInterface
 {
     private $cloner;
     private $dumper;
+    private $connection;
 
-    public function __construct(ClonerInterface $cloner, DataDumperInterface $dumper)
+    public function __construct(ClonerInterface $cloner, DataDumperInterface $dumper, Connection $connection = null)
     {
         $this->cloner = $cloner;
         $this->dumper = $dumper;
+        $this->connection = $connection;
     }
 
     public function configure()
     {
         $cloner = $this->cloner;
         $dumper = $this->dumper;
+        $connection = $this->connection;
 
-        VarDumper::setHandler(function ($var) use ($cloner, $dumper) {
-            $dumper->dump($cloner->cloneVar($var));
+        VarDumper::setHandler(static function ($var) use ($cloner, $dumper, $connection) {
+            $data = $cloner->cloneVar($var);
+
+            if ($connection) {
+                $connection->write($data);
+            }
+            $dumper->dump($data);
```

Comment:
is it an issue to always write to the local console even when there is a connection?